pkg/hostagent/hostagent.go

```diff
@@ -221,6 +223,9 @@ func determineSSHLocalPort(y *limayaml.LimaYAML, instName string) (int, error) {
 	if *y.SSH.LocalPort < 0 {
 		return 0, fmt.Errorf("invalid ssh local port %d", y.SSH.LocalPort)
 	}
+	if *y.SSH.LocalPort == 0 && *y.SSH.Address != "127.0.0.1" {
```

Comment:
https://pkg.go.dev/net#IP.IsLoopback should be used

Reply:
Here we are only checking if the YAML has been set, to something other than the default value.